test/blackbox-tests/test-cases/pkg/opam-repository-download.t

```diff
@@ -99,3 +100,65 @@ containing the repository:
   fake-xdg-cache/dune/opam-repository/packages/foo/foo.0.0.1
   fake-xdg-cache/dune/opam-repository/packages/foo/foo.0.0.1/opam
   fake-xdg-cache/dune/opam-repository/repo
+
+Let's update the repo with a stamp as an identifier of the repo state:
+
+  $ cat > mock-opam-repository/repo <<EOF
+  > opam-version: "2.0"
+  > stamp: "repository-identifier"
+  > EOF
+  $ cd mock-opam-repository
+  $ tar czf ../index.tar.gz *
+  $ cd ..
+
+  $ dune exec -- ./server.exe $PORT &
```

Comment:
Do you kill the server somewhere in this test?

Reply:
The server terminates automatically after serving one request, it's specifically written that way.